lib/modules/dosomething/dosomething_api/resources/kudos_resource.inc

```diff
@@ -145,9 +148,17 @@ function _kudos_resource_access($op) {
   }
 
   if ($op === 'create' || $op === 'delete') {
-    if (!user_is_anonymous()) {
-      return TRUE;
+    // Only auth'd users should be giving Kudos
+    if (user_is_anonymous()) {
+      return FALSE;
+    }
+
+    // We need to make sure people are only creating Kudos for themselves
+    if ($op === 'create' && $user->uid !== $args[1]) {
```

Comment:
I don't believe we're doing that anymore (neither mobile app nor Northstar has kudos). They'd also break regardless since the "API user" wouldn't have the same UID as the target kudos owner.